### PR TITLE
Do not read the whole file to compute SHA1 hash (fixes 15997)

### PR DESCRIPTION
**Pull request by irdassis**

### CODE REVIEW

File: lib/std/sha1.nim [hunk 1]
```diff
@@ -213,7 +213,22 @@ proc secureHashFile*(filename: string): SecureHash =
...
+  var state = newSha1State()
+  var buffer = newString(BufferLength)
+  while true:
+    let length = readChars(f, buffer, 0, BufferLength)
+    if length == 0:
+        break
+    buffer.setLen(length)
+    state.update(buffer)
+    if length != BufferLength:
+        buffer.setLen(BufferLength)
```
Comment (Araq) [inline lib/std/sha1.nim]:
```suggestion
      buffer.setLen(BufferLength)
```

File: lib/std/sha1.nim [hunk 2]
```diff
@@ -213,7 +213,22 @@ proc secureHashFile*(filename: string): SecureHash =
...
-  secureHash(readFile(filename))
+  const BufferLength = 8192
+
+  let f = open(filename)
+  var state = newSha1State()
+  var buffer = newString(BufferLength)
+  while true:
+    let length = readChars(f, buffer, 0, BufferLength)
+    if length == 0:
+        break
```
Comment (Araq) [inline lib/std/sha1.nim]:
```suggestion
      break
```

File: lib/std/sha1.nim (hunk 1 above)
Comment (Araq) [reply lib/std/sha1.nim]: And also: Why? ``if length != BufferLength`` you're done, might as well break the loop.

Comment (irdassis) [reply lib/std/sha1.nim]: True, I will change.

**Comment by timotheecour** [comment]
@irdassis as mentioned in General commit rules in https://nim-lang.github.io/Nim/contributing.html your PR (commit msg + body) should reference the issue via `fixes #15997` (note the pound #) otherwise the issue won't auto-close
(refs https://github.com/nim-lang/Nim/issues/15997#issuecomment-730078491)

**Pull request merged**